ecl/eclagent/eclgraph.cpp

```diff
@@ -1805,6 +1811,13 @@ void EclAgent::executeGraph(const char * graphName, bool realThor, size32_t pare
             }
 
             activeGraph.setown(loadGraph(graphName, queryWorkUnit(), dll, false));
+            unsigned guillotineCost = queryWorkUnit()->getDebugValueInt("maxCost", softCostLimit);
+            if (guillotineCost)
+            {
+                if (hardCostLimit && guillotineCost > hardCostLimit)
+                    guillotineCost = hardCostLimit;
+                abortmonitor->setGuillotineCost(guillotineCost);
```

Comment:
re. other question about whether guillotineCost can be 0.
Looks like if workunit setting "maxCost" could be 0, in which case abortmonitor->setGuillotineCost would not be called.
So I think the checkCostInterval logic and check needs to be disabled in that case.